setup.php

```diff
@@ -186,6 +186,31 @@ function plugin_init_genericobject() {
          PluginGenericobjectType::getType(),
          'getTypesForFormcreator'
       ];
+
+      // Add icon folder
+      if (!file_exists(GLPI_PLUGIN_DOC_DIR . "/genericobject/impact_icons/")) {
+         mkdir(GLPI_PLUGIN_DOC_DIR . "/genericobject/impact_icons/");
+      }
+
+      // Enable impact analysis
+      foreach (
+         (new PluginGenericobjectType())->find([
+            'use_impact' => true,
+         ]) as $row
+      ) {
+         $CFG_GLPI['impact_asset_types'][$row['itemtype']] = PluginGenericobjectType::getImpactIconFileStoragePath(
+            $row['impact_icon'],
+            true
+         );
+
+         // Update DB conf
+         $enabled = Config::getConfigurationValue('core', Impact::CONF_ENABLED);
+         $enabled = importArrayFromDB($enabled);
+         $enabled[] = $row['itemtype'];
+         Config::setConfigurationValues('core', [
+            Impact::CONF_ENABLED => exportArrayToDB($enabled)
+         ]);
```

Comment:
Doing it durign initialization of plugin is really dangerous.
Isn't it possible to only do it when the `use_impact` feature is enabled ?

Reply:
I can look into it, but what's the issue with doing this here ?

Reply:
1. It will not be possible to remove it from this list in GLPI configuration page, which may be considered as a bug.
2. Each time the plugin will be loaded, you will have to read, and potentially write the configuration. It will not be efficient at all.

Another possibility is to not add this `use_impact` field, adn to automatically push every genericobject itemtype in `$CFG_GLPI['impact_asset_types']`, so configuration will be made in existing configuration page.

Reply:
I think a5dc579 should be a good solution.
`use_impact` is no longer stored in the database, it just reflect whether or not the itemtype is enabled in the global impact configuration.
Changing the checkbox state will update the configuration accordingly.